trunk/src/main/srs_main_server.cpp

```diff
@@ -243,6 +243,7 @@ srs_error_t do_main(int argc, char** argv, char** envp)
 #endif
     
     if ((err = run_directly_or_daemon()) != srs_success) {
+        srs_free_global_system_ips();
```

Comment:
Please refine the code:

```cpp
    if ((err = run_directly_or_daemon()) != srs_success) {
        srs_free_global_system_ips();
        return srs_error_wrap(err, "run");
    }

    srs_free_global_system_ips();
```

to reduce some duplicated code by:

```cpp    
err = run_directly_or_daemon();
srs_free_global_system_ips();
if (err != srs_success) {
    return srs_error_wrap(err, "run");
}
```

Reply:
refine done